syntax = "proto3";
package distinct;
import "server/auth.proto";
service InventoryService {
  rpc Post(PostRequest) returns (StatusReply);
  rpc Get(GetRequest) returns (GetResponse);
}

message PostRequest {
  string token = 1;
  string type  = 2;
  string sku   = 3;
  int32 qty    = 4;
}

message GetRequest {
  string token = 1;
  string type  = 2;
  string sku   = 3;
}

message Item {
  string sku = 1;
  string name = 2;
  int32 stock = 3;
}

message GetResponse {
  string status = 1;
  repeated Item items = 2;
  string message = 3;
}
message VoteRequest {
  int32 term = 1;
  int32 candidateId = 2;
  int32 lastLogIndex = 3;
  int32 lastLogTerm = 4;
}

message VoteReply {
  int32 term = 1;
  bool voteGranted = 2;
}

message AppendRequest {
  int32 term = 1;
  int32 leaderId = 2;
  int32 prevLogIndex = 3;
  int32 prevLogTerm = 4;
  int32 commitIndex = 5;
  repeated string entries = 6;
}

message AppendReply {
  int32 term = 1;
  bool success = 2;
}

service RaftService {
  rpc RequestVote(VoteRequest) returns (VoteReply);
  rpc AppendEntries(AppendRequest) returns (AppendReply);
}